packages/atomic/src/components/search/atomic-generated-answer/atomic-generated-answer.tsx

```diff
@@ -143,30 +184,45 @@ export class AtomicGeneratedAnswer implements InitializableComponent {
             {this.bindings.i18n.t('generated-answer-title')}
           </Heading>
 
-          {!this.hasRetryableError && (
-            <div class="feedback-buttons flex gap-2 ml-auto">
-              <FeedbackButton
-                title={this.bindings.i18n.t('this-answer-was-helpful')}
-                variant="like"
-                active={this.generatedAnswerState.liked}
-                onClick={this.generatedAnswer.like}
-              />
-              <FeedbackButton
-                title={this.bindings.i18n.t('this-answer-was-not-helpful')}
-                variant="dislike"
-                active={this.generatedAnswerState.disliked}
-                onClick={this.generatedAnswer.dislike}
-              />
-            </div>
-          )}
+          <div class="flex gap-2 h-9 items-center ml-auto">
+            {!this.hasRetryableError && this.isAnswerVisible && (
+              <div class="feedback-buttons flex gap-2 ml-auto">
+                <FeedbackButton
+                  title={this.bindings.i18n.t('this-answer-was-helpful')}
+                  variant="like"
+                  active={this.generatedAnswerState.liked}
+                  onClick={this.generatedAnswer.like}
+                />
+                <FeedbackButton
+                  title={this.bindings.i18n.t('this-answer-was-not-helpful')}
+                  variant="dislike"
+                  active={this.generatedAnswerState.disliked}
+                  onClick={this.generatedAnswer.dislike}
+                />
+              </div>
+            )}
+
+            <Switch
+              part="toggle"
+              checked={this.isAnswerVisible}
+              onToggle={(checked) => {
+                checked
+                  ? this.generatedAnswer.show()
+                  : this.generatedAnswer.hide();
+              }}
+              ariaLabel={this.bindings.i18n.t('generated-answer-title')}
```

Comment:
Should the ariaLabel change based on the state of the switch? E.g., if checked is true, it would be "Hide generated answer" and if it's false it would be "Show generated answer"?

Reply:
According to [these guidelines](https://www.w3.org/WAI/ARIA/apg/patterns/switch/#:~:text=Important%3A%20it%20is%20critical%20the%20label%20on%20a%20switch%20does%20not%20change%20when%20its%20state%20changes.), the label should remain the same regardless of the switch state.

Reply:
I went with this as the switch label because it is the only piece of text that is visible when the generated answer is collapsed. Considering it acts like a section title, it made sense to me to also use it as the switch label.